src/Test/Perf/util/test_util.csx

```diff
@@ -12,125 +12,185 @@ using System.Threading;
-    Directory.CreateDirectory(path);
-    return path;
-}
+    public string DebugCscPath => Path.Combine(BinDebugDirectory, "csc.exe");
+    
+    public string ReleaseCscPath => Path.Combine(BinReleaseDirectory, "csc.exe");
 
-static string MyTempDirectory()
-{
-    var workingDir = MyWorkingDirectory();
-    var path = Path.Combine(workingDir, "temp");
-    Directory.CreateDirectory(path);
-    return path;
-}
+    public string DebugVbcPath => Path.Combine(BinDebugDirectory, "vbc.exe");
 
-static string RoslynDirectory()
-{
-    var workingDir = MyWorkingDirectory();
-    var srcTestPerf = Path.Combine("src", "Test", "Perf").ToString();
-    return workingDir.Substring(0, workingDir.IndexOf(srcTestPerf));
-}
+    public string ReleaseVbcPath => Path.Combine(BinReleaseDirectory, "vbc.exe");
+    
+    public string CPCDirectoryPath
+    {
+        get {
+            return Environment.ExpandEnvironmentVariables(@"%SYSTEMDRIVE%\CPC");
+        }
+    }
 
-static string PerfDirectory()
-{
-    return Path.Combine(RoslynDirectory(), "src", "Test", "Perf");
-}
+    public string GetViBenchToJsonExeFilePath => Path.Combine(CPCDirectoryPath, "ViBenchToJson.exe");
+    
+    
+    
+    /// Downloads a zip from azure store and extracts it into
+    /// the ./temp directory.
+    ///
+    /// If this current version has already been downloaded
+    /// and extracted, do nothing.
+    public void DownloadProject(string name, int version)
+    {
+        var zipFileName = $"{name}.{version}.zip";
+        var zipPath = Path.Combine(MyTempDirectory, zipFileName);
+        // If we've already downloaded the zip, assume that it
+        // has been downloaded *and* extracted.
+        if (File.Exists(zipPath))
+        {
+            Log($"Didn't download and extract {zipFileName} because one already exists.");
+            return;
+        }
 
-static string BinDirectory()
-{
-    return Path.Combine(RoslynDirectory(), "Binaries");
-}
+        // Remove all .zip files that were downloaded before.
+        foreach (var path in Directory.EnumerateFiles(MyTempDirectory, $"{name}.*.zip"))
+        {
+            Log($"Removing old zip {path}");
+            File.Delete(path);
+        }
 
-static string BinDebugDirectory()
-{
-    return Path.Combine(BinDirectory(), "Debug");
+        // Download zip file to temp directory
+        var downloadTarget = $"https://dotnetci.blob.core.windows.net/roslyn-perf/{zipFileName}";
+        Log($"Downloading {downloadTarget}");
+        var client = new WebClient();
+        client.DownloadFile(downloadTarget, zipPath);
+        Log($"Done Downloading");
+
+        // Extract to temp directory
+        Log($"Extracting {zipPath} to {MyTempDirectory}");
+        ZipFile.ExtractToDirectory(zipPath, MyTempDirectory);
+        Log($"Done Extracting");
+    }
 }
 
-static string BinReleaseDirectory()
-{
-    return Path.Combine(BinDirectory(), "Release");
+abstract class PerfTest: RelativeDirectory {
```

Comment:
I wanted people writing tests to be able to just write `MyTempDirectory` instead of using an accessor and then calling the method that they want.

Regardless, most of this will go away once we switch to putting everything in the binaries directory.